### PR TITLE
Add experimental method to create DO replicas

### PR DESCRIPTION
**Pull request by MellowYarker**

### CODE REVIEW

**Review by justin-mp — APPROVED** [review]

File: src/workerd/io/actor-cache.c++ [hunk 1]
```diff
@@ -3362,4 +3362,8 @@ kj::Promise<void> ActorCacheInterface::waitForBookmark(kj::StringPtr bookmark) {
       Error, "This Durable Object's storage back-end does not implement point-in-time recovery.");
 }
 
+void ActorCacheInterface::ensureReplicas() {
+  JSG_FAIL_REQUIRE(Error, "This Durable Object's storage back-end does not support replication.");
```
Comment (jasnell) [inline src/workerd/io/actor-cache.c++:3366]: Nit: Can you add a comment along the lines of this being intended to be implemented internally

File: src/workerd/io/actor-cache.h [hunk 2]
```diff
@@ -239,6 +239,8 @@ class ActorCacheInterface: public ActorCacheOps {
   virtual kj::Promise<kj::String> getBookmarkForTime(kj::Date timestamp);
   virtual kj::Promise<kj::String> onNextSessionRestoreBookmark(kj::StringPtr bookmark);
   virtual kj::Promise<void> waitForBookmark(kj::StringPtr bookmark);
+
+  virtual void ensureReplicas();
```
Comment (jasnell) [inline src/workerd/io/actor-cache.h:243]: Given that it's just a throw, consider just inlining the implementation with the throw here. Just keeps things a bit simpler.

**Review by jasnell — APPROVED** [review]

File: src/workerd/io/actor-cache.h (hunk 2 above)
Comment (MellowYarker) [reply src/workerd/io/actor-cache.h]: https://github.com/cloudflare/workerd/compare/9e46063539971abff34e8043709d95965cbec721..999831c699fd945aaddc50c446e07a6986d4a3b2

File: src/workerd/io/actor-cache.c++ (hunk 1 above)
Comment (MellowYarker) [reply src/workerd/io/actor-cache.c++]: https://github.com/cloudflare/workerd/compare/999831c699fd945aaddc50c446e07a6986d4a3b2..ced94358c1676c72530c24e3487170339ff98e8f

**Pull request merged**